### PR TITLE
Code refactor to support SM v2 and v3

### PR DESCRIPTION
**Pull request by ezimuel**
This PR contains the changes to support zend-servicemanager v2 and v3.

### CODE REVIEW

**Comment by ezimuel** [comment]
The travis is failing due to dependecy of zend-mvc that should be update for SM v2+v3.

File: .travis.yml [hunk 1]
```diff
@@ -17,13 +17,24 @@ matrix:
     - php: 5.5
       env:
         - EXECUTE_CS_CHECK=true
+    - php: 5.5
+      env:
+        - SERVICE_MANAGER_VERSION='>=2.7.3,<3.0'
```
Comment (weierophinney) [inline .travis.yml]: This can be more simply `^2.7.3`.

File: src/WriterPluginManager.php [hunk 2]
```diff
@@ -11,22 +11,79 @@
...
-        'xml'  => 'Zend\Config\Writer\Xml',
+    protected $instanceOf = Writer\AbstractWriter::class;
+
+    protected $aliases = [
+        'ini'      => Writer\Ini::class,
+        'Ini'      => Writer\Ini::class,
+        'json'     => Writer\Json::class,
+        'Json'     => Writer\Json::class,
+        'php'      => Writer\PhpArray::class,
+        'phpArray' => Writer\PhpArray::class,
```
Comment (weierophinney) [inline src/WriterPluginManager.php]: Need the lowercase version of this one, too.

Comment (ezimuel) [inline src/WriterPluginManager.php]: Fixed.

File: .travis.yml (hunk 1 above)
Comment (ezimuel) [inline .travis.yml]: Fixed

**Pull request merged**